extensions/Oversight/migrateToRevDel.php

```diff
@@ -0,0 +1,280 @@
+								'rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id )
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'rev_timestamp DESC' )
+						),
+						$dbw->selectRow(
+							'hidden',
+							array( 'rev_id' => 'hidden_rev_id', 'timestamp' => 'hidden_timestamp' ),
+							array(
+								'hidden_page' => $hiddenRow->hidden_page,
+								'hidden_timestamp < ' . $dbw->addQuotes( $hiddenRow->hidden_timestamp ),
+								'hidden_rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id )
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'hidden_timestamp DESC' )
+						),
+						$dbw->selectRow(
+							'archive',
+							array( 'rev_id' => 'ar_rev_id', 'timestamp' => 'ar_timestamp' ),
+							array(
+								'ar_page_id' => $hiddenRow->hidden_page,
+								'ar_timestamp < ' . $dbw->addQuotes( $hiddenRow->hidden_timestamp ),
+								'ar_rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id ),
+								'ar_namespace' => $hiddenRow->hidden_namespace,
+								'ar_title' => $hiddenRow->hidden_title
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'ar_timestamp DESC' )
+						)
+					);
+
+					$timestampsToRevIds = array();
+					foreach ( $parentIdFromTables as $parentIdFromTable ) {
+						if ( $parentIdFromTable != false ) {
+							$timestampsToRevIds[$parentIdFromTable->timestamp] = $parentIdFromTable->rev_id;
+						}
+					}
+
+					if ( count( $timestampsToRevIds ) == 0 ) {
+						$insertData['rev_parent_id'] = 0;
+						$this->output( "Warning: There may be an issue with revision ID {$hiddenRow->hidden_rev_id}. It has rev_parent_id=0 which means that it may be treated as a revision which starts a new page.\n" );
+					} else {
+						$highestTimestamp = max( array_keys( $timestampsToRevIds ) );
+						$insertData['rev_parent_id'] = $timestampsToRevIds[$highestTimestamp];
+					}
+				}
+
+				if ( $tableName == 'revision' ) {
+					$insertRevisionData[] = $insertData;
+				} else {
+					$insertArchiveData[] = $insertData;
+				}
+
+				$logData = array(
+					'log_type'      => 'suppress',
+					'log_action'    => 'revision',
+					'log_timestamp' => $hiddenRow->hidden_on_timestamp,
+					'log_user'      => $hiddenRow->hidden_by_user,
+					'log_user_text' => isset( $userNames[$hiddenRow->hidden_by_user] ) ? $userNames[$hiddenRow->hidden_by_user] : null,
+					'log_namespace' => $hiddenRow->hidden_namespace,
+					'log_title'     => $hiddenRow->hidden_title,
+					'log_page'      => $hiddenRow->hidden_page,
+					'log_comment'   => $hiddenRow->hidden_reason,
+					'log_params'    => "revision\n" . $hiddenRow->hidden_rev_id . "\nofield=" . $hiddenRow->hidden_deleted . "\nnfield=" . $insertData[$fieldPrefix . 'deleted'], //'revision', rev_id, old bits, new bits
+				);
+				if ( $logData['log_user_text'] === null ) {
+					throw new Exception( 'Unable to get user text for user ID ' . $hiddenRow->hidden_by_user . ' who hid revision ID ' . $hiddenRow->hidden_rev_id );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)